src/commcare_cloud/terraform/modules/ga_alb_waf/waflogs_add_partitions.py

```diff
@@ -0,0 +1,112 @@
+            print("Key is: ",key)
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == "404":
+            file_exists = False
+        else:
+            raise e
+    return file_exists
+
+
+def athena_query(session, params, wait = True):
+    client = session.client('athena')
+    
+    ## This function executes the query and returns the query execution ID
+    response_query_execution_id = client.start_query_execution(
+        QueryString = params['query'],
+        QueryExecutionContext = {
+            'Database' : "default"
+        }
+    )
+
+    if not wait:
+        return response_query_execution_id['QueryExecutionId']
+    else:
+        response_get_query_details = client.get_query_execution(
+            QueryExecutionId = response_query_execution_id['QueryExecutionId']
+        )
+        status = 'RUNNING'
+        iterations = 360 # 30 mins
+
+        while (iterations > 0):
+            iterations = iterations - 1
+            response_get_query_details = client.get_query_execution(
+            QueryExecutionId = response_query_execution_id['QueryExecutionId']
+            )
+            res = response_get_query_details['QueryExecution']
+            status = response_get_query_details['QueryExecution']['Status']['State']
+            print(status)
+            if (status == 'FAILED') or (status == 'CANCELLED') :
+                response = response_get_query_details['QueryExecution']['Status']['StateChangeReason']
+                return status, response
+
+            elif status == 'SUCCEEDED':
+                location = response_get_query_details['QueryExecution']['ResultConfiguration']['OutputLocation']
+
+                ## Function to get output results
+                response_query_result = client.get_query_results(
+                    QueryExecutionId = response_query_execution_id['QueryExecutionId']
+                )
+                result_data = response_query_result['ResultSet']
+
+                if len(response_query_result['ResultSet']['Rows']) > 1:
+                    header = response_query_result['ResultSet']['Rows'][0]
+                    rows = response_query_result['ResultSet']['Rows'][1:]
+                    return location, status
+                else:
+                    return location, "No rows"
+            else:
+                time.sleep(5)
+    return False
+
+def handler(event, context=None, params=None):
+    # get UTC date 
+    environment = os.environ['environment']
+    REGION = os.environ['AWS_REGION']
+    utc_datetime = datetime.utcnow()
+    year = utc_datetime.strftime('%Y')
+    month = utc_datetime.strftime('%m')
+    day = utc_datetime.strftime("%d")
+    hour = utc_datetime.strftime('%H')
+
+    DATABASE = 'default'
+    TABLE = 'waf_logs'
+
+    # S3 constant
+    S3_OUTPUT = f"s3://dimagi-commcare-{environment}-logs"
+    S3_BUCKET = f"dimagi-commcare-{environment}-logs"
+    STREAM_PATH=f"frontend-waf-partitioned-{environment}"
```

Comment:
Are these the only lines that differ from the other similar lambda for formplayer request response logs? I wonder if there is a way to share code between the two. Is there a way to have a single lambda that can be called with different environment variables? If so then maybe these can be environment variables

Reply:
@dannyroberts sure. will look into it.

Reply:
@dannyroberts if we use this Partition projection feature from AWS, then we don't need a Lambda function to add new partitions. I'm looking into this. Would like to try this on Staging for waf logs, Please let me  know your thoughts.

https://docs.aws.amazon.com/athena/latest/ug/waf-logs.html#create-waf-table-partition-projection

Reply:
Yeah this sounds great, and it sounds like it's working out. I'll wait until you ping me again before reviewing further then.

Reply:
Because I would imagine can delete this lambda now right?

Reply:
Yes, Danny, we don't need that for waf logs partitioning. 
For Formplayer logs I will create a separate ticket to create the existing table and delete lambda function.